src/modules/Dropdown/Dropdown.js

```diff
@@ -949,6 +962,17 @@ export default class Dropdown extends Component {
   }
 
   // ----------------------------------------
+  // Overrides
+  // ----------------------------------------
+
+  handleIconOverrides = predefinedProps => ({
+    onClick: (e) => {
+      _.invoke(predefinedProps, 'onClick', e, predefinedProps)
```

Comment:
I see the issue here, however, I think this is going to be rare enough of an edge case to leave as-is for now. We shall see :)